Java/core/src/main/java/com/amazon/randomcutforest/store/INodeStore.java

```diff
@@ -143,4 +143,24 @@
      * @return
      */
     int getMaxIndex();
+
+    /**
+     * increases the mass of node as well as all its ancestors by 1 note that all
+     * these nodes are internal nodes and we eliminate the back and forth in a
+     * single call
+     * 
+     * @param index node
+     */
+    void increaseMassOfAncestorsAndItselfRecursively(int index);
```

Comment:
Ok - agree with the incrementMass. But for decrement, we check for 0 (at leaf) and only proceed one node at a time primarily because the delete is simpler. Do we want to say adjustMass for the increment?